provider/lxd/provider.go

```diff
@@ -395,3 +363,115 @@ type interfaceAddress struct{}
 func (interfaceAddress) InterfaceAddress(interfaceName string) (string, error) {
 	return utils.GetAddressForInterface(interfaceName)
 }
+
+type serverFactory struct{}
+
+func (serverFactory) LocalServer() (Server, error) {
```

Comment:
Is it still possible to instantiate a local server more than one in the course of bootstrapping localhost?

If it is, I would cache the server as a member of ```serverFactory``` and return that value for future calls.

Reply:
This won't work because of the bionic issue, where we have to ask for a new server connection for the local address.

Reply:
https://bugs.launchpad.net/juju/+bug/1771887

Reply:
But could we push that check loop here? I think it's reasonable to consider supply of a ready-to-use local connection a concern of the factory.

This method that does the verification and attendant set-up is implicitly a one-use method, that might not be treated as such by someone in the future.